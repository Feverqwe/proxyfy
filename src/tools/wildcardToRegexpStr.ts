import escapeStringRegexp from "escape-string-regexp";

function wildcardToRegexpStr(pattern: string) {
  let patterns = [];
  const m = /^(.+\/)?([^\/]+)$/.exec(pattern);
  if (m) {
    const scheme = m[1] || '';
    const hostname = m[2];

    patterns.push(scheme + hostname);
    if (/^\*\./.test(hostname)) {
      patterns.push(scheme + hostname.substr(2));
    }
  } else {
    patterns.push(pattern);
  }

  return patterns.map((pattern) => {
    const re = escapeStringRegexp(pattern).replace(/\\([*?])/g, '.$1');
    return `^${re}$`;
  });
}

export default wildcardToRegexpStr;
