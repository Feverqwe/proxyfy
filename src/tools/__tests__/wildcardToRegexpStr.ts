import wildcardToRegexpStr from "../wildcardToRegexpStr";

describe('wildcardToRegexpStr', () => {
  test('a', () => {
    const result = wildcardToRegexpStr('*.test.com');

    console.log(result);

    expect(result).toMatchSnapshot();
  });

  test('b', () => {
    const result = wildcardToRegexpStr('*://*.test.com');

    console.log(result);

    expect(result).toMatchSnapshot();
  });

  test('c', () => {
    const result = wildcardToRegexpStr('*://*test.com');

    console.log(result);

    expect(result).toMatchSnapshot();
  });

  test('d', () => {
    const result = wildcardToRegexpStr('*test.com');

    console.log(result);

    expect(result).toMatchSnapshot();
  });
});
