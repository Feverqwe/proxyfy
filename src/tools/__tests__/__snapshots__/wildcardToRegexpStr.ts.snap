// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wildcardToRegexpStr a 1`] = `
Array [
  "^.*\\\\.test\\\\.com$",
  "^test\\\\.com$",
]
`;

exports[`wildcardToRegexpStr b 1`] = `
Array [
  "^.*://.*\\\\.test\\\\.com$",
  "^.*://test\\\\.com$",
]
`;

exports[`wildcardToRegexpStr c 1`] = `
Array [
  "^.*://.*test\\\\.com$",
]
`;

exports[`wildcardToRegexpStr d 1`] = `
Array [
  "^.*test\\\\.com$",
]
`;
